extensions/modules/file/src/test/resources/util/helper.xqm

```diff
@@ -0,0 +1,192 @@
+(:
+ : eXist-db Open Source Native XML Database
+ : Copyright (C) 2001 The eXist-db Authors
+ :
+ : info@exist-db.org
+ : http://www.exist-db.org
+ :
+ : This library is free software; you can redistribute it and/or
+ : modify it under the terms of the GNU Lesser General Public
+ : License as published by the Free Software Foundation; either
+ : version 2.1 of the License, or (at your option) any later version.
+ :
+ : This library is distributed in the hope that it will be useful,
+ : but WITHOUT ANY WARRANTY; without even the implied warranty of
+ : MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ : Lesser General Public License for more details.
+ :
+ : You should have received a copy of the GNU Lesser General Public
+ : License along with this library; if not, write to the Free Software
+ : Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+ :)
+xquery version "3.1";
+
+module namespace helper="http://exist-db.org/xquery/test/util/helper";
+import module namespace fixtures="http://exist-db.org/xquery/test/util/fixtures" at "fixtures.xqm";
+import module namespace file="http://exist-db.org/xquery/file";
+import module namespace xmldb="http://exist-db.org/xquery/xmldb";
+import module namespace util="http://exist-db.org/xquery/util";
+
+declare variable $helper:error := xs:QName("helper:assert-sync-error");
+
+(:
+/db
+    /file-module-test
+        /data
+            test-data.xml
+        test-text.txt
+        test-query.xq
+        bin
+:)
+declare function helper:setup-db() as empty-sequence() {
+    (
```

Comment:
The way that this is structured will make Java debugging of such XQuery much more difficult than it needs to be. Ideally we don't want to involve anything clever (such as the Arrow Operator) or superfluous in a test (unless the test is actually for the Arrow operator). We also should avoid any unnecessary operators and function calls. As such, I would instead suggest this should be refactored to the much simpler XQuery 1.0 code:

```xquery
let $_ := (
        xmldb:create-collection("/db", $fixtures:collection-name),
        helper:create-db-resource($fixtures:collection, "test-text.txt", $fixtures:TXT),
        helper:create-db-resource($fixtures:collection, "test-query.xq", $fixtures:XQY),
        helper:create-db-resource($fixtures:collection, "bin", $fixtures:BIN),

        xmldb:create-collection($fixtures:collection, $fixtures:child-collection-name),
        helper:create-db-resource($fixtures:child-collection, "test-data.xml", $fixtures:XML)
    )
   return ()
```
This will make Java debugging of the running XQuery much simpler. It will also as an unintended side-effect speed up the execution of this code.